Fields listed after a field of type `thread` in struct generates a C error.

### Describe the bug

Placing struct fields (of any type it seems) after `field_name thread string` produces a C error.

### Reproduction Steps

```v
// Video is the object containing all the information needed for
// conversion to ascii.
struct Video {
pub:
	// path may be URL or path on local file system.
	path         string
	tmp_dir      string
	frame_count  int
	frame_format FrameFormat
pub mut:
	// TODO: once V fixes bug #19281 replace the type here
	// with `thread !`. And the return type on the corresponding
	// function that is spawned.
	extract_frames_thread thread string
	frames                [][]u8
	framerate             f64 = 24.0
}

enum FrameFormat {
	bmp
	jpg
}

fn (video Video) extract_frames() string {
	// try to extract frames os.execute()
	// if os.execute() returns with an exit
	// code other than 0, return an error.
	if true {
		return 'Failed to extract frames: os.execute() cmd output'
	}
	// do something else
	return ''
}

fn main() {
	mut video := Video{
		path: '/path/to/some/mp4'
		tmp_dir: '/tmp'
	}
	video.extract_frames_thread = spawn video.extract_frames()
}
```

### Expected Behavior

Expected code to compile and run with no output.

### Current Behavior

```
==================
C:/Users/imado/AppData/Local/Temp/v_0/test.8575829116616490619.tmp.c:6854: warning: implicit declaration of function 'tcc_backtrace'
C:/Users/imado/AppData/Local/Temp/v_0/test.8575829116616490619.tmp.c:12633: error: field expected
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V full version: V 0.4.1 639f128

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.1 639f128
OS: windows, Microsoft Windows 11 Pro v22621 64-bit
Processor: 16 cpus, 64bit, little endian, 

getwd: C:\Users\imado\Documents\test
vexe: C:\Users\imado\v\v.exe
vexe mtime: 2023-09-05 17:00:49

vroot: OK, value: C:\Users\imado\v
VMODULES: OK, value: C:\Users\imado\.vmodules
VTMP: OK, value: C:\Users\imado\AppData\Local\Temp\v_0

Git version: git version 2.33.1.windows.1
Git vroot status: 0.4.1-10-g639f128c
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,
operable program or batch file.

thirdparty/tcc status: thirdparty-windows-amd64 e90c2620
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -6302,7 +6302,7 @@ fn (mut g Gen) type_default(typ_ ast.Type) string {
 		.string {
 			return '(string){.str=(byteptr)"", .is_lit=1}'
 		}
-		.interface_, .sum_type, .array_fixed, .multi_return {
+		.interface_, .sum_type, .array_fixed, .multi_return, .thread {
 			return '{0}'
 		}
 		.alias {

--- a/vlib/v/slow_tests/inout/printing_struct_with_thread_field.out
+++ b/vlib/v/slow_tests/inout/printing_struct_with_thread_field.out
@@ -4,4 +4,6 @@ Video{
     frame_count: 0
     frame_format: bmp
     extract_frames_thread: thread(string)
+    frames: []
+    framerate: 24.0
 }

--- a/vlib/v/slow_tests/inout/printing_struct_with_thread_field.vv
+++ b/vlib/v/slow_tests/inout/printing_struct_with_thread_field.vv
@@ -12,6 +12,8 @@ pub mut:
 	// with `thread !`. And the return type on the corresponding
 	// function that is spawned.
 	extract_frames_thread thread string
+	frames                [][]u8
+	framerate             f64 = 24.0
 }
 
 enum FrameFormat {

